simulator.py

```diff
@@ -94,13 +94,34 @@ def __init__(self, env, name, age, rng, infection_timestamp, household, workplac
         self.avg_misc_time = _draw_random_discreet_gaussian(AVG_MISC_MINUTES, SCALE_MISC_MINUTES, self.rng)
         self.scale_misc_time = _draw_random_discreet_gaussian(AVG_SCALE_MISC_MINUTES, SCALE_SCALE_MISC_MINUTES, self.rng)
 
-        # TODO: multiple possible days and times & limit these activities in a week
-        self.shopping_days = self.rng.choice(range(7))
-        self.shopping_hours = self.rng.choice(range(7, 20))
-
-        self.exercise_days = self.rng.choice(range(7))
-        self.exercise_hours = self.rng.choice(range(7, 20))
-
+        #getting the number of shopping days and hours from a distribution
+        self.number_of_shopping_days = _draw_random_discreet_gaussian(AVG_NUM_SHOPPING_DAYS, SCALE_NUM_SHOPPING_DAYS, self.rng)
+        self.number_of_shopping_hours = _draw_random_discreet_gaussian(AVG_NUM_SHOPPING_HOURS, SCALE_NUM_SHOPPING_HOURS, self.rng)
+
+        #getting the number of exercise days and hours from a distribution
+        self.number_of_exercise_days = _draw_random_discreet_gaussian(AVG_NUM_EXERCISE_DAYS, SCALE_NUM_EXERCISE_DAYS, self.rng)
+        self.number_of_exercise_hours = _draw_random_discreet_gaussian(AVG_NUM_EXERCISE_HOURS, SCALE_NUM_EXERCISE_HOURS, self.rng)
+
+        #Multiple shopping days and hours
+        self.shopping_days = self.rng.choice(range(7), self.number_of_shopping_days)
+        self.shopping_hours = self.rng.choice(range(7, 20), self.number_of_shopping_hours)
+
+        #Multiple exercise days and hours
+        self.exercise_days = self.rng.choice(range(7), self.number_of_exercise_days)
+        self.exercise_hours = self.rng.choice(range(7, 20), self.number_of_exercise_hours)
+
+        #Limiting the number of hours spent shopping per week
+        self.avg_max_shop_per_week = _draw_random_discreet_gaussian(AVG_NUM_SHOP_PER_WEEK, SCALE_NUM_SHOP_PER_WEEK, self.rng)
+        self.scale_max_shop_per_week = _draw_random_discreet_gaussian(AVG_SCALE_NUM_SHOP_PER_WEEK, SCALE_SCALE_NUM_SHOP_PER_WEEK, self.rng)
```

Comment:
It might be less confusing to have `AVG_MAX_NUM_SHOP_PER_WEEK` rather than `AVG_NUM_SHOP_PER_WEEK`. You don't need to draw a scale (line 115) because you are not drawing a maximum every week. There is no flaw in drawing a number every week, but it might not make a difference.

Reply:
I've made those changes and committed it to my master branch